modules/programs/hexchat.nix

```diff
@@ -0,0 +1,323 @@
+            commands = [
+              "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+            ];
+            loginMethod = sasl;
+            nickname = "my_nickname";
+            nickname2 = "my_secondchoice";
+            options = {
+              acceptInvalidSSLCertificates = false;
+              autoconnect = true;
+              bypassProxy = true;
+              connectToSelectedServerOnly = true;
+              useGlobalUserInformation = false;
+              forceSSL = false;
+            };
+            password = "my_password";
+            realName = "my_realname";
+            servers = [
+              "chat.freenode.net"
+              "irc.freenode.net"
+            ];
+            userName = "my_username";
+          };
+        }'';
+    };
+
+    settings = mkOption {
+      default = null;
+      description = ''
+        Configuration for "~/.config/hexchat/hexchat.conf", see
+        <link xlink:href="https://hexchat.readthedocs.io/en/latest/settings.html#list-of-settings"/>
+        for supported values.
+      '';
+      example = literalExample ''
+        {
+          irc_nick1 = "mynick";
+          irc_username = "bob";
+          irc_realname = "Bart Simpson";
+          text_font = "Monospace 14";
+        };
+      '';
+      type = nullOr (attrsOf str);
+    };
+
+    overwriteConfigFiles = mkOption {
+      type = nullOr bool;
+      description = ''
+        Enables overwritting HexChat configuration files (hexchat.conf, servlist.conf).
+        Any existing HexChat configuration will be lost.
+        Certify to back-up any previous configuration before enabling this.
+
+        Enabling this setting is recommended, because everytime HexChat application is
+        closed it overwrites Nix/Home-Manager provided configuration files, causing:
+        1. Nix/HM provided configuration to be out of sync with actual active HexChat configuration.
+        2. Blocking Nix/HM updates until configuration files are manually removed.
+      '';
+      default = false;
+    };
+
+    theme = mkOption {
+      default = null;
+      description = ''
+        Theme package for HexChat.
+        Expects a derivation containing decompressed theme files.
+        '.hct' file format requires unzip decompression, as seen in example.
+      '';
+      example = ''
+        stdenv.mkDerivation rec {
+          name = "hexchat-theme-MatriY";
+          buildInputs = [ pkgs.unzip ];
+          src = fetchurl {
+              url = "https://dl.hexchat.net/themes/MatriY.hct";
+              sha256 = "sha256-ffkFJvySfl0Hwja3y7XCiNJceUrGvlEoEm97eYNMTZc=";
+          };
+          unpackPhase = "unzip ''${src}";
+          installPhase = "cp -r . $out";
+        };
+      '';
+      type = nullOr package;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.hexchat ];
+    xdg.configFile."hexchat" = mkIf (cfg.theme != null) {
+      source = cfg.theme;
+      recursive = true;
+    };
```

Comment:
Themes are located at the root of the hexchat configuration folder?

Reply:
I think so, but @superherointj can confirm.

Reply:
Yes, themes are located at: `~/.config/hexchat`.
It `overwrites` files there: colors.conf  '[Content_Types].xml'